data/transactions/logic/eval.go

```diff
@@ -982,7 +990,7 @@ func (cx *evalContext) txnFieldToStack(txn *transactions.Transaction, field uint
 	case LastValid:
 		sv.Uint = uint64(txn.LastValid)
 	case Note:
-		sv.Bytes = txn.Note
+		sv.Bytes = nilToEmpty(txn.Note)
```

Comment:
this makes it a slightly larger change, but could we instead have a method `SetBytes` on `(sv *stackValue)` that internally calls `sv.Bytes = nilToEmpty(b)`? That way if we add new byte slice fields we can't fail to call `nilToEmpty`

Reply:
(in place of doing `sv.Bytes =` everywhere in this function)